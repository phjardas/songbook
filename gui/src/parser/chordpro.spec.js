import fs from 'fs';
import path from 'path';
import { promisify } from 'util';
import parse from './chordpro';

const readFile = promisify(fs.readFile);

async function parseSong(name) {
  const file = path.resolve(__dirname, `chordpro.spec.${name}.txt`);
  const content = await readFile(file, 'utf8');
  return parse(content);
}

describe('parser', () => {
  describe('chordpro', () => {
    it('should parse a song', async () => {
      const song = await parseSong('song1');
      expect(song).toMatchSnapshot();
    });
  });
});
