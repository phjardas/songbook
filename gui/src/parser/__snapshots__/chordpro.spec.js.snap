// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser chordpro should parse a song 1`] = `
Object {
  "messages": Array [
    Object {
      "line": 2,
      "message": "Ignoring directive \\"title\\"",
      "type": "warning",
    },
  ],
  "sections": Array [
    Object {
      "lines": Array [
        Object {
          "parts": Array [
            Object {
              "text": "I ",
            },
            Object {
              "chord": "D",
              "text": "looked over Jordan, and ",
            },
            Object {
              "chord": "G",
              "text": "what did I ",
            },
            Object {
              "chord": "D",
              "text": "see,",
            },
          ],
        },
        Object {
          "parts": Array [
            Object {
              "text": "Comin’ for to carry me ",
            },
            Object {
              "chord": "A7",
              "text": "home.",
            },
          ],
        },
      ],
    },
    Object {
      "label": "Chorus",
      "lines": Array [
        Object {
          "parts": Array [
            Object {
              "text": "Swing ",
            },
            Object {
              "chord": "D",
              "text": "low, sweet ",
            },
            Object {
              "chord": "G",
              "text": "chari",
            },
            Object {
              "chord": "D",
              "text": "ot,",
            },
          ],
        },
        Object {
          "parts": Array [
            Object {
              "text": "Comin’ for to carry me ",
            },
            Object {
              "chord": "A7",
              "text": "home.",
            },
          ],
        },
      ],
      "type": "chorus",
    },
    Object {
      "label": "Verse 2",
      "lines": Array [
        Object {
          "parts": Array [
            Object {
              "text": "I ",
            },
            Object {
              "chord": "D",
              "text": "looked over Jordan, and ",
            },
            Object {
              "chord": "G",
              "text": "what did I ",
            },
            Object {
              "chord": "D",
              "text": "see,",
            },
          ],
        },
        Object {
          "parts": Array [
            Object {
              "text": "Comin’ for to carry me ",
            },
            Object {
              "chord": "A7",
              "text": "home.",
            },
          ],
        },
      ],
      "type": "verse",
    },
    Object {
      "lines": Array [
        Object {
          "parts": Array [
            Object {
              "text": "I ",
            },
            Object {
              "chord": "D",
              "text": "looked over Jordan, and ",
            },
            Object {
              "chord": "G",
              "text": "what did I ",
            },
            Object {
              "chord": "D",
              "text": "see,",
            },
          ],
        },
        Object {
          "parts": Array [
            Object {
              "text": "Comin’ for to carry me ",
            },
            Object {
              "chord": "A7",
              "text": "home.",
            },
          ],
        },
      ],
    },
  ],
}
`;
